src/test/scala/com/github/mrpowers/spark/fast/tests/DataFrameComparerTest.scala

```diff
@@ -254,6 +253,135 @@ class DataFrameComparerTest extends FreeSpec with DataFrameComparer with SparkSe
       }
     }
 
-  }
+    "can performed unordered schema DataFrame comparisons" in {
+      val sourceDF = spark.createDF(
+        List(
+          (1, "word"),
+          (5, "word")
+        ),
+        List(
+          ("number", IntegerType, true),
+          ("word", StringType, true)
+        )
+      )
+      val expectedDF = spark.createDF(
+        List(
+          ("word", 1),
+          ("word", 5)
+        ),
+        List(
+          ("word", StringType, true),
+          ("number", IntegerType, true)
+        )
+      )
+      assertSmallDataFrameEquality(sourceDF, expectedDF, ignoreColumnOrder = true)
+    }
+
+    "can performed unordered schema and unordered row DataFrame comparisons" in {
+      val sourceDF = spark.createDF(
+        List(
+          (5, "word"),
+          (1, "word")
+        ),
+        List(
+          ("number", IntegerType, true),
+          ("word", StringType, true)
+        )
+      )
+      val expectedDF = spark.createDF(
+        List(
+          ("word", 1),
+          ("word", 5)
+        ),
+        List(
+          ("word", StringType, true),
+          ("number", IntegerType, true)
+        )
+      )
+      assertSmallDataFrameEquality(sourceDF, expectedDF, orderedComparison = false, ignoreColumnOrder = true)
+    }
+
+    "throws an error for unordered column DataFrame comparisons that don't match" in {
+      val sourceDF = spark.createDF(
+        List(
+          (5, "word"),
+          (5, "word")
+        ),
+        List(
+          ("number", IntegerType, true),
+          ("word", StringType, true)
+        )
+      )
+      val expectedDF = spark.createDF(
+        List(
+          ("word", 1),
+          ("word", 5)
+        ),
+        List(
+          ("word", StringType, true),
+          ("number", IntegerType, true)
+        )
+      )
+      val e = intercept[DatasetContentMismatch] {
+        assertSmallDataFrameEquality(sourceDF, expectedDF, ignoreColumnOrder = true)
+      }
```

Comment:
It may be worth asserting a portion of the error message from the `DataContentMismatch` exception. Doing so ensures that the error thrown relates to an unordered column comparison issue and not some unexpected issue of the same error type.